Source/Shared/arcana/functional/inplace_function.h

```diff
@@ -362,8 +362,15 @@ namespace stdext
                 break;
             case Operation::Copy:
             {
-                const FunctorT* source = (const FunctorT*)const_cast<void*>(fromPtr);
-                new (thisFunctor) FunctorT(*source);
+                if constexpr (std::is_copy_constructible<FunctorT>::value)
+                {
+                    const FunctorT* source = (const FunctorT*)const_cast<void*>(fromPtr);
+                    new (thisFunctor) FunctorT(*source);
+                }
+                else
+                {
+                    throw std::runtime_error("Cannot copy non-copyable inplace function.");
```

Comment:
We could `static_assert` a `bool` here (set via compile flag or otherwise) to have the option to enable or
disable non-copyable inplace functions here. I went back and forth on this, as it seems suboptimal to have this type of problem be a runtime error by default, but adding a mechanism to switch between compile time and runtime failure for our one particular use case seemed like complexity for complexity's sake. It's not a heavyweight mechanism, though: just a single `#ifdef` controlling the definition of a private `static constexpr bool` in this class. Thoughts on whether that's worth making part of this change?